src/javascript/JContent/ContentRoute/ContentLayout/Upload/UploadItem/UploadItem.jsx

```diff
@@ -189,37 +190,46 @@ export class UploadItem extends React.Component {
         }
 
         if (type === 'replace') {
+            if (window.contextJsParameters.config.links.tagImageOnUpload) {
```

Comment:
Using the registry can be ok, but it must be generic and not introduce a dependency to automated tag module. You can simple use a registry.find({type:'fileUpload'}), iterate on the results and take the first callback that can handle he upload. For more clarity I would replace all 4 client.mutate here by registry entries, and completely replace the handleUpload method by just :   
```
    handleUpload(type) {
        return registry.find({type: 'fileUpload'}).find(upload => upload.canHandle(type, this.props)).handleUpload(type, this.props);
    }
```

The "default" fileupload handlers would be registered by jcontent initialization by something like : 
```
        registry.add('fileUpload', 'import', {
            canHandle: type => type === 'import',
            handleUpload: (type, {path, file, client}) => (
                client.mutate({
                    mutation: importContent,
                    variables: {
                        path: path,
                        fileHandle: file
                    }
                })
            )
        });
```

I did not check the previous PR, but all reference to AutomatedTags must be removed from jcontent.

Reply:
Thanks @tdraier I see what you mean (appreciated for the code snippet)